src/main/java/winstone/accesslog/SimpleAccessLogger.java

```diff
@@ -95,44 +97,39 @@ public SimpleAccessLogger(String webAppName, Map<String, String> startupArgs) th
 
     @Override
     public void log(Request request, Response response) {
-        String uriLine = request.getMethod() + " " + request.getRequestURI() + " " + request.getProtocol();
+        String uriLine = request.getMethod() + " " + request.getHttpURI().getPath() + " "
+                + request.getConnectionMetaData().getProtocol();
         int status = response.getStatus();
-        long size = response.getContentCount();
+        long size = Response.getContentBytesWritten(response);
         String date;
         synchronized (DF) {
             date = DF.format(new Date());
         }
-
+        Request.AuthenticationState authenticationState = Request.getAuthenticationState(request);
+        Principal principal = authenticationState == null ? null : authenticationState.getUserPrincipal();
+        String remoteUser = principal == null ? "-" : principal.getName();
         // mimic
-        // https://github.com/eclipse/jetty.project/blob/jetty-9.4.0.v20161208/jetty-server/src/main/java/org/eclipse/jetty/server/AbstractNCSARequestLog.java#L130
-        Authentication authentication = request.getAuthentication();
-        String remoteUser;
-        if (authentication instanceof Authentication.User) {
-            Authentication.User user = (Authentication.User) authentication;
-            remoteUser = user.getUserIdentity().getUserPrincipal().getName();
-        } else {
-            remoteUser = null;
-        }
-
+        // https://github.com/jetty/jetty.project/blob/3632a57f2796a2ab4fcdbfe62837d494bcd4e94a/jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/CustomRequestLog.java#L307
```

Comment:
Given the content the comment is referring to, the comment is placed above the wrong block of code. It should be placed above `Request.AuthenticationState authenticationState`.

Reply:
the link is going to the NCSA format used by the Jetty.

Reply:
In the old (currently shipping) code, there was a single comment above the copy-pasted code to set the remote user, showing where it was originally from. The new version still has copy-pasted code to set the remote user, but has lost any reference to the original code, which is technically a regression (in documentation). To fix this, the original link should be changed from a reference to the copy-pasted Jetty 10 code to the copy-pasted Jetty 12 code (https://github.com/jetty/jetty.project/blob/c5b2533fdecce21b54c6fbaf36f79bc3ba909775/jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/CustomRequestLog.java#L1093-L1099).

Adding a link to the NCSA format is a valuable change, but it isn't strictly related to this PR (moving from Jetty 10 to Jetty 12), so I think if that change is desired then a separate PR should be opened to introduce it. That PR could be merged immediately and would reduce the size of this diff.